pkgpanda/util.py

```diff
@@ -264,13 +266,28 @@ def make_file(name):
 
 
 def write_json(filename, data):
-    with open(filename, "w+") as f:
-        return json.dump(data, f, **json_prettyprint_args)
+    dumped_json = json.dump(data, f, **json_prettyprint_args)
+    write_string(filename, dumped_json)
+    return dumped_json
 
 
 def write_string(filename, data):
-    with open(filename, "w+") as f:
-        return f.write(data)
+    """
+    Write a string to a file.
+    Overwrite any data in that file.
+
+    We use an atomic write practice of creating a temporary file and then
+    moving that temporary file to the given ``filename``. This prevents race
+    conditions such as the file being read by another process after it is
+    opened here but not yet written to.
```

Comment:
Do you think we can also mention here truncating a file in a 'full disk' scenario?

Reply:
Done below.